### DISCUSSION TITLE
unable to receive sign message using signMessageAsync

### DISCUSSION THREAD

**Discussion by eric-ccw** [wevm/wagmi#2696] [Q&A]
Hi there, the message is not received right after the user logged in. However, I received the message when connect it again in the second time. Is there any way to get the message after logged in? This happened in walletConnectConnector only.

my code
```javascript
let walletAddress
const { isConnected, address } = useAccount()
const { connectAsync } = useConnect()
const { signMessageAsync } = useSignMessage()

if (isConnected) {
walletAddress = address
} else {
const { account } = await connectAsync({ connector })
walletAddress = account
}

const siweMessage = new SiweMessage({
    domain: window.location.host,
    address: walletAddress,
    statement: 'Sign in with Ethereum to the app.',
    uri,
    version: '1',
    chainId,
    nonce,
})

const signature = await signMessageAsync({
  message: siweMessage.prepareMessage(),
})
```

**Comment by SaadNiaziDev** [comment]
Use sign message in callback of useAccount. Check for isReconnected flag as wagmi reconnects when page loads
`const account = useAccount({
    onConnect({ address, connector, isReconnected }) {
      if(!isReconnected ) signMessage()
    },
  })`

**Reply by adrianotadao** [reply]
Helo there,

quick question for you.... when I call signMessage, metamask is not being called... what could it be? ideas?

**Reply by SaadNiaziDev** [reply]
Maybe you have multiple extensions or either metamask is disabled/locked. Try to manually unlock and sign message , see if it works or not.

**Reply by adrianotadao** [reply]
hey @SaadNiaziDev thank you for the reply.

long story short, I see the message to sign on metamask but I need to manually go there to sign... different from when I am connecting I am redirected to metamask automatically

I think I understood what is going on. I am using it on a React native app so it seems that web3modal + walletconnect is losing the connection (or something like this) with metamask and this is why it does not redirect to metamask while signing the message. 

Let me investigate it.

**Reply by adrianotadao** [reply]
related https://github.com/wevm/wagmi/discussions/1128#discussioncomment-7975768
